terraform/gitlab/gitlab.tf.json.template.py

```diff
@@ -532,9 +533,7 @@ def remove_inconsequential_statements(statements: List[JSON]) -> List[JSON]:
         "aws_vpc": {
             "gitlab": {
                 "cidr_block": vpc_cidr,
-                "tags": {
-                    "Name": "azul-gitlab"
-                }
+                "tags": tags("azul-gitlab", component="azul-gitlab")
```

Comment:
In my brief test, calling `config.unqualified_resource_name_or_none('azul-gitlab')` returns `(None, None)`.

Reply:
Yes, this is one of the trickier parts of the functionality. You're right that `config.unqualified_resource_name_or_none` (and by extension, `tags()`) will choke on `azul-gitlab` as-is. My workaround is to specify the component manually -- `tags()` won't overwrite an explicitly provided tag, so after post-processing is complete the tags are "correct." Without the workaround, `component` will be `azul-None`.

This definitely isn't preferred behavior, since we want to avoid specifying things manually if we can. Is there a better way to approach this case? I initially thought this was the best approach, but reconsidering it now I don't think it is clear to someone reading the code.

Reply:
I see. I don't really have a better idea.

Maybe you could assert that if an unqualified name cannot be parsed, that the component tag is also passed in as an arguement. This might help prevent accidental misuse in the future.

Reply:
I thought about this over the weekend and I agree with that solution -- this case shouldn't fail silently.